fastlane/lib/fastlane/actions/sonar.rb

```diff
@@ -19,7 +21,7 @@ def self.run(params)
         sonar_scanner_args << "-Dsonar.exclusions=\"#{params[:exclusions]}\"" if params[:exclusions]
         sonar_scanner_args << "-Dsonar.language=\"#{params[:project_language]}\"" if params[:project_language]
         sonar_scanner_args << "-Dsonar.sourceEncoding=\"#{params[:source_encoding]}\"" if params[:source_encoding]
-        sonar_scanner_args << "-Dsonar.login=\"#{params[:sonar_login]}\"" if params[:sonar_login]
+        sonar_scanner_args << "-Dsonar.token=\"#{sonar_token_arg}\"" if sonar_token_arg
```

Comment:
If you mean by using sonar_login in the Fastfile or when using the sonar action by fastlane yes. At line 13 we take either the old or the new parameter and we pass it to the sonar-scanner CLI tool.

If you are asking if we support both the old SonarQube server (on premise and cloud), before they added the sonar.token parameter, and the newer CLI where they did add it and then deprecated the previous option, then no... Should fastlane support the old deprecated SonarQube and the new one? I am not sure if we should and how to go about it.  The .token option was added in the 10.0 release which went public in March 2023. 

<img width="685" alt="Screenshot 2023-12-22 at 15 37 37" src="https://github.com/fastlane/fastlane/assets/562991/b4e1ee98-4f9c-4853-a54a-f558aa193213">

Reply:
Until it becomes an error to specify the -Dsonar.login parameter, we could add both -Dsonar.login and -Dsonar.token, but then we still get the warning... so this PR becomes a bit useless :D. Until, we fully move to the new syntax... Or we move to the new syntax (8 months or so ago new :)).

Reply:
Until it becomes an error to specify the -Dsonar.login parameter, we could add both -Dsonar.login and -Dsonar.token, but then we still get the warning... so this PR becomes a bit useless :D. Until, we fully move to the new syntax... Or we move to the new syntax (8 months or so ago new :)).

Reply:
@Panajev Hmm my point here is that with this PR, if there's anyone out there using the old (deprecated) way, once they upgrade fastlane, their setup will break. Thus, this would be a breaking change, in a minor (not major) fastlane version bump 😅 

I think we'd better keep the old implementation in place, as well as allow for the newer one, like this:

```diff
-sonar_token_arg = params[:sonar_token] || params[:sonar_login]
```

```ruby
sonar_scanner_args << "-Dsonar.login=\"#{params[:sonar_login]}\"" if params[:sonar_login]
sonar_scanner_args << "-Dsonar.token=\"#{params[:sonar_token]}\"" if params[:sonar_token]
```

Since these 2 parameters (`sonar_login` and `sonar_token`) are mutually exclusive (guaranteed at runtime in the validation step), then we have the guarantee that only 1 of the 2 sonar scanner args will be included, hence keep supporting both the older (deprecated) and the newer syntax.

Does this make sense?

Reply:
@rogerluan fine by me :). I implemented your suggestion. I was not sure for how long the old param should be kept around since the LTS version it is offered in is over 2.5 years old and its LTS support terminated already. 

When would we deprecate this old parameter? When the SonarQube 10 LTS is out?